src/Response/ResponseExtensions.cs

```diff
@@ -58,5 +63,118 @@ public static class ResponseExtensions
+
+            if (contentDisposition != null)
+            {
+                response.Headers["Content-Disposition"] = contentDisposition.ToString();
+            }
+
+            // rangeHeader should be of the format "bytes=0-" or "bytes=0-12345" or "bytes=123-456"
+            var rangeHeader = response.HttpContext.Request.Headers["Range"].FirstOrDefault();
+
+            if (!string.IsNullOrEmpty(rangeHeader) && rangeHeader.Contains("="))
+            {
+                var rangeParts = rangeHeader.SplitOnFirst("=")[1].SplitOnFirst("-");
+                rangeStart = long.Parse(rangeParts[0]);
+                rangeEnd = rangeParts.Length == 2 && !string.IsNullOrEmpty(rangeParts[1])
+                    ? int.Parse(rangeParts[1]) // the client requested a chunk
+                    : contentLength - 1;
+
+                if (rangeStart < 0 || rangeEnd > contentLength - 1)
+                {
+                    response.StatusCode = (int)HttpStatusCode.RequestedRangeNotSatisfiable;
+                    await response.WriteAsync(string.Empty);
+                }
+                else
+                {
+                    response.Headers["ContentRange"] = $"bytes {rangeStart}-{rangeEnd}/{contentLength}";
+                    response.StatusCode = (int)HttpStatusCode.PartialContent;
+                    await GetResponseBodyDelegate(rangeStart, rangeEnd, contentLength, source, response.Body, response.HttpContext.RequestAborted);
+                }
+            }
+            else
+            {
+                rangeStart = 0;
+                rangeEnd = contentLength - 1;
+                await GetResponseBodyDelegate(rangeStart, rangeEnd, contentLength, source, response.Body, response.HttpContext.RequestAborted);
+            }
+        }
+
+        private static async Task GetResponseBodyDelegate(long rangeStart, long rangeEnd, long contentLength, Stream source, Stream destination, CancellationToken cancellationToken)
+        {
+            if (rangeStart == 0 && rangeEnd == contentLength - 1)
+            {
+                await StreamCopyOperation.CopyToAsync(source, destination, new long?(), 65536, cancellationToken);
+            }
+            else
+            {
+                if (!source.CanSeek)
+                {
+                    throw new InvalidOperationException("Sending Range Responses requires a seekable stream eg. FileStream or MemoryStream");
+                }
+
+                var totalBytesToSend = rangeEnd - rangeStart + 1;
+                const int BufferSize = 0x1000;
+                var buffer = new byte[BufferSize];
+                var bytesRemaining = totalBytesToSend;
+
+                source.Seek(rangeStart, SeekOrigin.Begin);
+                while (bytesRemaining > 0)
+                {
+                    var count = bytesRemaining <= buffer.Length
+                        ? await source.ReadAsync(buffer, 0, (int)Math.Min(bytesRemaining, int.MaxValue), cancellationToken)
```

Comment:
`Math.Min(bytesRemaining, int.MaxValue)` should never return something other than `bytesRemaining`.

You probably meant to do this to avoid the ternary?
```
var count = await source.ReadAsync(buffer, 0, (int)Math.Min(bytesRemaining, buffer.Length), cancellationToken);
```